lib/guess_html_encoding.rb

```diff
@@ -19,11 +19,20 @@ def self.guess(html, headers = nil)
     end
 
     if out.nil? || out.empty? || !encoding_loaded?(out)
-      if html =~ /<meta[^>]*HTTP-EQUIV=["']?Content-Type["']?[^>]*content=["']([^'"]*)["']/i && $1 =~ /charset=([\w\d-]+);?/i
-        out = $1
-      elsif html =~ /<meta\s+charset=["']([\w\d-]+)?/i
-        out = $1
+
+      meta_tags_regex = /
+      (?:
+        <meta[^>]*HTTP-EQUIV=["']?Content-Type["']?[^>]*content=["'][^'"]*charset\s*=\s*([^"'\;\s]+)  # http-equiv
```

Comment:
@cantino bit confused by what you mean here – a `['"]` after `content=["'][^'"]*charset\s*=\s*` doesn't make sense to me, given that we're matching the value of the complete `content=` attribute here (the `charset` attribute is handled by the line below).

I'll have a look around and see if I can find any existing lists of valid and invalid test cases...